net/http/src/THttpWSEngine.cxx

```diff
@@ -26,9 +26,9 @@
 ////////////////////////////////////////////////////////////////////////////////
 /// Attach WSEngine to THttpCallArg to transport to the WSHandler
 
-void THttpWSEngine::AttachTo(THttpCallArg &arg)
+THttpWSEngine::THttpWSEngine(std::shared_ptr<THttpCallArg> arg)
 {
-   arg.SetWSEngine(this);
+   arg->fWSEngine.reset(this); // set ownership of created
```

Comment:
Could you make this `THttpCallArg::CreateEngine()` or similar? This pattern (constructor taking its owner) is so rare that it's surprising me again and again. It's just really bad to have `new` without assigning the result.

Reply:
Main problem - I do not like to make this methods public.
THttpCallArg is central class of http requests processing and one should not provide easy methods to confuse users.

Reply:
Ok, I create THttpCallArg::CreateWSEngine method. Here is commit:

https://github.com/linev/root/commit/09fcb4ceb7f7ba81fac3dc4a2a230b1eefd7f424

I will submit it with other PR